### PR TITLE
Switched btree dep to boxpack

### LINKED ISSUE #6: Consider Removing binpacking Dep
`binpacking` has an issue with module exports. - https://github.com/jsmarkus/node-bin-packing/issues/3

This makes running spritesmith (or layout) in an environment like atom/node-webkit impossible.

The current version is three years old with no active maintainer, would you consider forking the corresponding btree code and crediting the author in this repo?

I'd be more than happy to throw in a PR if you are willing to merge.

### PR DESCRIPTION
**Pull request by jonathandelgado**
Fixes #6 

Great suggestion on `boxpack`, really easy switch.

All of the iterations for the input and output properties are no longer necessary, the new module doesn't use the naming shortcuts the old one did.

Tests are passing. I also regenerated a handful of my project spritesheets, all were identical through the switch, but please double check before merging, you probably know of a few edge cases?

### CODE REVIEW

File: lib/algorithms/binary-tree.algorithm.js [hunk 1]
```diff
@@ -11,30 +10,10 @@ exports.sort = function (items) {
...
-    item.x = fit.x;
-    item.y = fit.y;
-    delete item.fit;
-  });
+  // Create packer instance
+  var packer = new boxpack();
 
   // Return the packed items
-  return items;
+  return packer.pack(items);
```
Comment (twolfson) [inline lib/algorithms/binary-tree.algorithm.js]: Damn, that is way cleaner

**Comment by twolfson** [comment]
Everything looks great at a glance. I will review more in depth tonight. Thanks for the fast turnaround!

**Comment by twolfson** [comment]
It looks like we are dropping metadata but I should have maintained that in the test suite =/

Going to take a look at the `boxpack` source. Worst case: We add metadata back after the algorithm runs.

**Comment by jonathandelgado** [comment]
@twolfson Good catch, didn't notice that.

Any suggestion on doing so? I assume the array order may not be persistent?

**Comment by twolfson** [comment]
Ah, it looks like it is persisted but hidden via `Object.create`:

https://github.com/munro/boxpack/blob/3809aa551a7603f2fee7496fcdaabc4c270d8ddd/boxpack.js#L71-L75

When I log the `meta` property directly, it appears. I am going to tack on another commit to add tests and re-expose the `meta` key as an enumerable property.

Thanks again!

**Comment by jonathandelgado** [comment]
@twolfson Phenomenal. Thank you for your quick replies!

**Comment by twolfson** [comment]
Thanks for your quick replies as well =)

This has been merged and released in `2.1.0`. I will make updates/releases in the `spritesmith` family shortly.

**Comment by twolfson** [comment]
Alright, the family has been upgraded:

- `spritesmith@1.1.0`
- `grunt-spritesmith@3.6.0`
- `gulp.spritsmith@2.5.0`

**Comment by jonathandelgado** [comment]
Awesome, thanks again! I really appreciate it.
